contracts/azorius/LinearERC721Voting.sol

```diff
@@ -0,0 +1,292 @@
+            }
+        }
+
+        GovernanceNFT memory governance = GovernanceNFT({
+            token: token,
+            weight: _weight,
+            isProposer: _isProposer
+        });
+
+        governanceTokens.push(governance);
+        emit GovernanceTokenAdded(_nftAddress);
+    }
+
+    function removeGovernanceToken(address _nftAddress) external onlyOwner {
+        if (governanceTokens.length == 0) revert LastGovernanceToken();
+
+        for (uint256 i = 0; i < governanceTokens.length; i++) {
+            if (address(governanceTokens[i].token) == _nftAddress) {
+                
+                uint256 last = governanceTokens.length - 1;
+                governanceTokens[i] = governanceTokens[last]; // move the last token into the position to remove
+                delete governanceTokens[last];                // delete the last token
+
+                emit GovernanceTokenRemoved(_nftAddress);
+                break;
+            }
+        }
+    }
+
+    /**
+     * Returns whether an NFT has already voted.
+     */
+    function hasVoted(uint32 _proposalId, address _tokenAddress, uint256 _tokenId) public view returns (bool) {
+        return proposalVotes[_proposalId].hasVoted[_tokenAddress][_tokenId];
+    }
+
+    /** @inheritdoc BaseStrategy*/
+    function isPassed(uint32 _proposalId) public view override returns (bool) {
+        return (
+            block.number > proposalVotes[_proposalId].votingEndBlock && // voting period has ended
+            quorumThreshold <= proposalVotes[_proposalId].yesVotes + proposalVotes[_proposalId].abstainVotes && // yes + abstain votes meets the quorum
+            meetsBasis(proposalVotes[_proposalId].yesVotes, proposalVotes[_proposalId].noVotes) // yes votes meets the basis
+        );
+    }
+
+    /** @inheritdoc BaseStrategy*/
+    function isProposer(address _address) public view override returns (bool) {
+        for (uint i = 0; i < governanceTokens.length; i++) {
+            GovernanceNFT memory governance = governanceTokens[i];
+            if (governance.isProposer && governance.token.balanceOf(_address) > 0) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /** @inheritdoc BaseStrategy*/
+    function votingEndBlock(uint32 _proposalId) public view override returns (uint32) {
+      return proposalVotes[_proposalId].votingEndBlock;
+    }
+
+    /** Internal implementation of `updateVotingPeriod`. */
+    function _updateVotingPeriod(uint32 _votingPeriod) internal {
+        votingPeriod = _votingPeriod;
+        emit VotingPeriodUpdated(_votingPeriod);
+    }
+
+    /** Internal implementation of `updateQuorumThreshold`. */
+    function _updateQuorumThreshold(uint256 _quorumThreshold) internal {
+        quorumThreshold = _quorumThreshold;
+        emit QuorumThresholdUpdated(quorumThreshold);
+    }
+
+    function _vote(uint32 _proposalId, address _voter, uint8 _voteType, uint256 weight) internal {
```

Comment:
If the `getWeight` function is changed so that it only takes a single token address / ID, the weight lookup could be done inside this function, and weight wouldn't be passed to this function as an argument.